File: leetcode/C++/patching-array.cpp
class Solution {
public:
    int minPatches(vector<int>& nums, int n) {
        long sol=0; 
        for(long avail=0, idx=0; avail<n;){
            if(idx<nums.size() && avail+1>=nums[idx]) avail+=nums[idx++];
            else{avail=2*avail+1; ++sol;}
        }
        return sol;
    }
};

File: leetcode/Python/patching-array.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        sol, avail, idx = 0, 0, 0
        while avail<n:
            if idx<len(nums) and avail+1>=nums[idx]:
                avail, idx = avail+nums[idx], idx+1
            else:
                avail, sol = 2*avail+1, sol+1
        return sol